### PR TITLE
codeintel: Wrap interfaces in observable structs

### PR DESCRIPTION
**Pull request by efritz**
Wrap each codeintel interface in a struct that emits observed operations (tracing, metrics, logging).

### CODE REVIEW

File: cmd/precise-code-intel-api-server/main.go [hunk 1]
```diff
@@ -26,18 +31,28 @@ func main() {
...
+	logger := log15.Root()
+	tracer := trace.Tracer{Tracer: opentracing.GlobalTracer()}
+	dbSpecs := db.NewSpecs(logger, tracer, "precise_code_intel_api_server")
+	codeIntelAPISpecs := api.NewSpecs(logger, tracer, "precise_code_intel_api_server")
+	dbSpecs.MustRegister(prometheus.DefaultRegisterer)
+	codeIntelAPISpecs.MustRegister(prometheus.DefaultRegisterer)
+
+	db := db.NewObservedDB(mustInitializeDatabase(), dbSpecs)
+	bundleManagerClient := bundles.New(bundleManagerURL)
+	codeIntelAPI := api.NewObservedCodeIntelAPI(api.New(db, bundleManagerClient), codeIntelAPISpecs)
```
Comment (gbrik) [inline cmd/precise-code-intel-api-server/main.go:48]: suggestion:
mismatched name format, can it be either `New` and `NewObserved`, or `NewCodeIntelAPI` and `NewObservedCodeIntelAPI`

Comment (efritz) [reply cmd/precise-code-intel-api-server/main.go:48]: Will update before merging.

**Comment by gbrik** [comment]
has anything changed except renaming since my last review? having trouble following the commits

**Comment by efritz** [comment]
@gbrik No it's all just merging with master and fixing conflicts.

**Pull request merged**